pkg/crd/persistentip/v1alpha1/types.go

```diff
@@ -0,0 +1,48 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//go:generate go run sigs.k8s.io/controller-tools/cmd/controller-gen@v0.13.0 paths=./... object output:dir=./
+
+//go:generate go run k8s.io/code-generator/cmd/client-gen@v0.28.0 client-gen --go-header-file ../../../../hack/custom-boilerplate.go.txt --clientset-name versioned --input-base "" --input github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1 --output-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/clientset ..
+
+//go:generate go run k8s.io/code-generator/cmd/lister-gen@v0.28.0 lister-gen --go-header-file ../../../../hack/custom-boilerplate.go.txt --input-dirs github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1 --output-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/listers ..
+
+//go:generate go run k8s.io/code-generator/cmd/informer-gen@v0.28.0 informer-gen --go-header-file ../../../../hack/custom-boilerplate.go.txt --input-dirs github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1 --versioned-clientset-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/clientset/versioned --listers-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/listers --output-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/informers ..
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=persistentips,shortName=pip,singular=persistentip,scope=Namespaced
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PersistentIP is the Schema for the PersistentIP API
+type PersistentIP struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PersistentIPSpec   `json:"spec,omitempty"`
+	Status PersistentIPStatus `json:"status,omitempty"`
```

Comment:
Do we have to mark this as subresource ?

Reply:
Probably ! 

Let me see what happens when I do so. 

Good catch !!!

Reply:
I actually thought that was done by the kubebuilder tag `// +kubebuilder:subresource:status`. 

Let me check the documentation ...

Reply:
Ahh wait didn't saw the tag, all is fine.